### PR TITLE
This introduces `thread_auto_close` and `thread_auto_close_response`.…

### LINKED ISSUE #161: suggestion: auto close ticket after specific time
Most companys support ticket system including discord close the ticket automatically if the user doesnt respond **after** moderator sent a reply within  a specific timeframe normally within 48 hours. 

And allow us to edit a custom message for bot to  send to the recipient explaining why the ticket got closed.
Perhaps allow modmail to just that aswell?

### PR DESCRIPTION
**Pull request by daegontaven**
… Both these can be used to configure modmail to auto close a thread after some time has passed.

`thread_auto_close_response` has a `%t` variable that can be used to insert human friendly time into the closing message.

This does not auto close threads that already exist and will have to be manually closed. But once the variable is configured, it will take it from there for every new thread.

**Take a look at this P.R with two sets of eyes as this is my first attempt at calling methods that I did not write.**

Resolves #161

### CODE REVIEW

File: core/thread.py [hunk 1]
```diff
@@ -336,6 +343,72 @@ def _close_after(self, closer, silent, delete_channel, message):
                 if str(message_id) == str(embed.author.url).split("/")[-1]:
                     return msg
 
+    async def _grab_timeout(
+            self
+    ) -> typing.Union[None, isodate.duration.Duration, timedelta]:
```
Comment (fourjr) [inline core/thread.py]: do this instead in one line:
```python
async def _grab_timeout(self) -> typing.Union[None, isodate.duration.Duration, timedelta]:
```

Also, rename the function to `_fetch_timeout` 😛

File: core/thread.py [hunk 2]
```diff
@@ -336,6 +343,72 @@ def _close_after(self, closer, silent, delete_channel, message):
...
+        seconds = timeout.total_seconds()
+        # seconds = 20  # Uncomment to debug with just 20 seconds
+        reset_time = datetime.utcnow() + timedelta(seconds=seconds)
+        human_time = human_timedelta(dt=reset_time)
+
+        # Grab message
+        close_message = self.bot.config.get(
+            "thread_auto_close_response",
+            f"This thread has been closed automatically after no response from"
+            f" you for {human_time}."
```
Comment (fourjr) [inline core/thread.py]: Use this message instead `This thread has been closed automatically due to inactivity after {human_time}`

File: core/thread.py (hunk 1 above)
Comment (daegontaven) [reply core/thread.py]: I was following PEP8. Are we using a different code style? 😅

File: core/thread.py (hunk 2 above)
Comment (daegontaven) [reply core/thread.py]: I'll push that commit when I get back home. Thanks!

File: core/thread.py (hunk 1 above)
Comment (fourjr) [reply core/thread.py]: We follow our code style lul

whatever looks/works best i guess..

**Comment by kyb3r** [comment]
Just use black for code formatting.

File: core/thread.py (hunk 1 above)
Comment (kyb3r) [reply core/thread.py]: Just use black for code formatting.

**Comment by kyb3r** [comment]
Can you also reflect the changes in the changelog? Thanks :)

File: core/thread.py (hunk 1 above)
Comment (daegontaven) [reply core/thread.py]: @fourjr I committed that into one line, but the second black formatting commit reverted it back.

**Pull request merged**